### ISSUE TITLE
FEATURE REQUEST: ini_manage should able to synchronize ini file from sections dict

### ISSUE THREAD

**Issue by tiadobatima** [saltstack/salt#29287]
Right now, If a certain option is set in the ini file, but it doesn't exist in the "sections dict" in salt.states.ini_manage.options_present, that option will remain unchanged:

From the docs:

```
options present in file and not specified in sections dict will be untouched
changes dict will contain the list of changes made
```

It would be great if there was a separate state function (or a switch in salt.states.ini_manage.options_present) that ensures that only options in the "sections dict"  are kept in the ini file. If an option doesn't exist in the "sections dict", it should be deleted from the ini file. Basically, synchronize the ini file with the "sections dict".

I'm not asking for the current behaviour to be changed, but both behaviours should be supported.

https://docs.saltstack.com/en/latest/ref/states/all/salt.states.ini_manage.html

Thanks!!!
g.

**Comment by jfindlay** [comment]
@tiadobatima, thanks for the feature request.

**Comment by Akilesh1597** [comment]
Can we add a new argument 'strict' to options_present and sections_present which would make sure that the ini file strictly resembles the data provided ?

**Comment by jfindlay** [comment]
@Akilesh1597, good suggestion.

**Comment by sevagh** [comment]
Hello, I took a stab at implementing this.

I largely focused on making the real state work correctly (in a Docker container).

The unit test for this module seems weird to me, and doesn't actually reflect the way data is passed from the real sls file to the state. I did what needed to be done to make the unit tests pass.

```
# cat test.ini

[test1]
a = a
b = b

[test2]
a = a
b = b
c = c
root@28d80e067092:/etc/salt/states/test# cat test_ini.sls 
/etc/salt/states/test/test.ini:
  ini.options_present:
    - separator: '='
    - strict: True
    - sections:
        test1:
          a: 'new_a'
root@28d80e067092:/etc/salt/states/test# salt-call state.sls test.test_ini
local:
----------
          ID: /etc/salt/states/test/test.ini
    Function: ini.options_present
      Result: True
     Comment: Changes take effect
     Started: 16:51:00.182237
    Duration: 2.189 ms
     Changes:   
              ----------
              test1:
                  ----------
                  a:
                      ----------
                      after:
                          new_a
                      before:
                          a
                  b:
                      ----------
                      after:
                          None
                      before:
                          b

Summary for local
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
Total run time:   2.189 ms
root@28d80e067092:/etc/salt/states/test# 
root@28d80e067092:/etc/salt/states/test# cat test.ini

[test1]
a = new_a

[test2]
a = a
b = b
c = c
root@28d80e067092:/etc/salt/states/test# 
```

**Comment by lorengordon** [comment]
This can probably be closed, as #38414 was merged and `strict` is available. @Ch3LL?
